torch/autograd/profiler.py

```diff
@@ -206,9 +205,8 @@ def __init__(
         self.enabled: bool = enabled
         if not self.enabled:
             return
-        self.use_cuda = use_cuda
         self.use_device: Optional[str] = (
-            use_device if use_device != "privateuseone" else None
+            None if use_device == "privateuseone" else use_device
```

Comment:
@fwenguang could privateuseone keep self.use_device="privateuseone"? Why do we set it to None?

Reply:
keep self.use_device="privateuseone" is okey to me, if so, some code below also need to be updated. For example, in some if conditional statements.

In fact, self.use_device="privateuseone" is equivalent to self.use_device=None  for our custom backend, since we will rename backend name to another string.

Reply:
`        if use_cuda:
            warn("The attribute `use_cuda` will be deprecated soon, please use ``use_device = 'cuda'`` instead.")
            self.use_device = "cuda"
        else:
            self.use_device: Optional[str] = (
                "privateuseone" if use_device is None else use_device
            )`

Reply:
OK, i change the code to below. When users set the use_cude, the use_device='cuda' and a warning is thrown. When users set the use_device, the use_device is set to what user set.
```
if use_cuda:
    warn("The attribute `use_cuda` will be deprecated soon, please use ``use_device = 'cuda'`` instead.")
    self.use_device = "cuda"
else:
    self.use_device: Optional[str] = (
        "privateuseone" if use_device is None else use_device
    )
```

Reply:
warning for use_cuda looks good to me. Ithink for self.use_device, it should:
```
self.use_device: Optional[str] = use_device
```